gq/gq.go

```diff
@@ -86,3 +90,17 @@ func randomBytes(rng io.Reader, byteCount int) ([]byte, error) {
 
 	return bytes, nil
 }
+
+func parseJWT(jwt []byte) ([]byte, []byte, error) {
+	headers, payload, signature, err := jws.SplitCompact(jwt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Signatures are over header and payload in the base64 url-encoded byte
+	// form of `header + '.' + payload`
+	signingPayload := append(headers, []byte(".")...)
```

Comment:
The main reason I prefer `append` is that it doesn't return an error (maybe because type-checking is stronger there?). `bytes.Buffer` is pretty clear, but we drop the error that `WriteX` returns currently and I think that will look messier with proper error handling. `append` to me seems like the more "fool-proof" approach but lmk, I'm fine turning it back to `bytes.Buffer`. Hopefully, this is code people will rarely touch so it probably doesn't matter as much.

Reply:
I think either is fine :smile: The `WriteX` functions on `bytes.Buffer` can't actually error, I think the errors in the return values are just for conformance to `io.Writer` etc

Reply:
gotcha, I'll stick with `append` then